### PR TITLE
Pre first release cleanup

### PR DESCRIPTION
**Pull request by skyw**
* Slightly updated test.
* Force keyword arguments for optimizers that have very long argument list, otherwise error prone
* Abstract weight decay to single mixin class
* Some code style fix/improvement

@mkhona-nvidia @FDecaYed please take a look, there will be another doc only clean up before we making a release, after which things changing things would become maintenance burden.

@mkhona-nvidia coverage of scalar optimizers and soap with adaptative criteria is low, would be good to improve. It can also be done after release, so not mandatory.

### CODE REVIEW

File: emerging_optimizers/orthogonalized_optimizers/orthogonalized_optimizer.py [hunk 1]
```diff
@@ -26,23 +26,23 @@
...
-            default to be False.
+        use_nesterov: Whether to use Nesterov-style momentum in the internal SGD.
+        weight_decay_method: Method to apply weight decay, see :class:`~emerging_optimizers.mixin.WeightDecayMixin`
+            for more details.
         fp32_matmul_prec: Precision of the matmul operations in optimizer states GEMM operations.
 """
 
 
-class OrthogonalizedOptimizer(optim.Optimizer):
+class OrthogonalizedOptimizer(opt_mixin.WeightDecayMixin, optim.Optimizer):
```
Comment (mkhona-nvidia) [inline emerging_optimizers/orthogonalized_optimizers/orthogonalized_optimizer.py:45]: Double inheritance makes this complicated

File: emerging_optimizers/mixin.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+from typing import Literal
+
+import torch
+
+
+WeightDecayT = Literal["decoupled", "independent", "l2"]
+
+
+class WeightDecayMixin:
```
Comment (mkhona-nvidia) [inline emerging_optimizers/mixin.py:24]: What is the reason for this to be a class rather than a set of functions that are chosen based on arguments?

File: emerging_optimizers/orthogonalized_optimizers/orthogonalized_optimizer.py (hunk 1 above)
Comment (skyw) [reply emerging_optimizers/orthogonalized_optimizers/orthogonalized_optimizer.py:45]: No, this is mixin. The only type of multi inheritance should be used in Python.

File: emerging_optimizers/mixin.py (hunk 2 above)
Comment (skyw) [reply emerging_optimizers/mixin.py:24]: weight decay is a function highly coupled with optimizer, and shared among a lot of optim subclass.

Thought about a optimizer based class with this function and make everyone inherit from it, but not all of our optimizers use the same base.

File: emerging_optimizers/orthogonalized_optimizers/scion.py [hunk 3]
```diff
@@ -69,14 +70,9 @@ def __init__(
             raise ValueError(f"num_ns_steps must be at least 1, got {num_ns_steps}")
 
         # Add checks for weight decay arguments to enable Franke-Wolfe step.
-        logging.info("Scion does not use weight decay. Setting weight_decay to 1.")
+        logging.info("Scion does not use weight decay. Setting weight_decay to 1 and weight_decay_method to l2.")
```
Comment (mkhona-nvidia) [inline emerging_optimizers/orthogonalized_optimizers/scion.py]: it should be `decoupled` for scion

Comment (skyw) [reply emerging_optimizers/orthogonalized_optimizers/scion.py]: oops...fixed.

**Review by FDecaYed — COMMENTED** [review]
LGTM
